servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/single/SingleConcatWithPublisherTest.java

```diff
@@ -60,7 +62,12 @@ class SingleConcatWithPublisherTest {
     private TestPublisher<Integer> next = new TestPublisher.Builder<Integer>().disableAutoOnSubscribe().build();
 
     void setUp(boolean deferSubscribe) {
-        toSource(source.concat(next, deferSubscribe)).subscribe(subscriber);
+        setUp(deferSubscribe, false);
+    }
+
+    void setUp(boolean deferSubscribe, boolean propagateCancel) {
+        toSource(propagateCancel ? source.concatPropagateCancel(next) : source.concat(next, deferSubscribe))
```

Comment:
Tests will be a bit misleading. They assume 4 combinations, but we have only 3 combinations. Let's try to avoid testing with `true, true` as it's the same as `false, true`.